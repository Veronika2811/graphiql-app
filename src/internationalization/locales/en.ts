import { TranslationKeys } from 'internationalization/TranslationKeys';

export const englishDictionary = {
  [TranslationKeys.signIn]: 'Sign In',
  [TranslationKeys.signUp]: 'Sign Up',
  [TranslationKeys.logOut]: 'Log Out',
  [TranslationKeys.not_found_page]: 'Not Found Page',
  [TranslationKeys.welcome_page]: 'Welcome Page',
  [TranslationKeys.main_title]: 'The Best IDE for GraphQL Requests',
  [TranslationKeys.main_subtitle]:
    'The app is the result of completing the final task of RS School React Course.',
  [TranslationKeys.welcome_link_editor_heading]:
    'Query and Explore GraphQL API with',
  [TranslationKeys.welcome_link_authorization_heading]: 'to get started',
  [TranslationKeys.about_project_title_1]: 'A query language for your API',
  [TranslationKeys.about_project_subtitle_1]:
    'GraphQL is a query language for APIs and a runtime for fulfilling those queries with your existing data. GraphQL provides a complete and understandable description of the data in your API, gives clients the power to ask for exactly what they need and nothing more, makes it easier to evolve APIs over time, and enables powerful developer tools.',
  [TranslationKeys.about_project_title_2]:
    'Ask for what you need,get exactly that',
  [TranslationKeys.about_project_subtitle_2]:
    'Send a GraphQL query to your API and get exactly what you need, nothing more and nothing less. GraphQL queries always return predictable results. Apps using GraphQL are fast and stable because they control the data they get, not the server.',
  [TranslationKeys.about_project_title_3]: 'About the RS School',
  [TranslationKeys.about_project_subtitle_3]:
    'RS School is free-of-charge and community-based education program conducted by The Rolling Scopes developer community since 2013. Everyone can study at RS School, regardless of age, professional employment, or place of residence. The mentors and trainers of school are front-end and javascript developers from different companies and countries.',
  [TranslationKeys.about_project_alt]: 'Section image',
  [TranslationKeys.team]: 'Team',
  [TranslationKeys.team_alt]: 'Profile picture',
  [TranslationKeys.editor_search_bar_label]: 'URL API',
  [TranslationKeys.editor_button_change_endpoint]: 'Change Endpoint',
  [TranslationKeys.konstantin]: 'Konstantin',
  [TranslationKeys.veranika]: 'Veranika',
  [TranslationKeys.artem]: 'Artem',
  [TranslationKeys.auth_title]: 'Welcome',
  [TranslationKeys.auth_email]: 'Email',
  [TranslationKeys.auth_password]: 'Password',
  [TranslationKeys.auth_confirm_password]: 'Confirm Password',
  [TranslationKeys.auth_name]: 'Name',
  [TranslationKeys.variables]: 'Variables',
  [TranslationKeys.headers]: 'Headers',
  [TranslationKeys.editor_message_missing_URL]: 'Enter endpoint',
  [TranslationKeys.editor_message_incorrect_URL]: 'Enter correct endpoint',
  [TranslationKeys.editor_message_missing_request]: 'Enter request',
  [TranslationKeys.editor_message_check_request]: 'Check request',
  [TranslationKeys.editor_message_successful_request]: 'Successful request',
  [TranslationKeys.loading]: 'Loading',
  [TranslationKeys.error_boundary_message]: 'Oops! Something went wrong!',
  [TranslationKeys.button_execute_query]: 'Execute query',
  [TranslationKeys.button_make_beautiful]: 'Make beautiful',
  [TranslationKeys.auth_success_sign_in]: 'Logged in successfully',
  [TranslationKeys.auth_success_sign_up]: 'Registered successfully',
  [TranslationKeys.dic_error_api_wrong_password]: 'Wrong password',
  [TranslationKeys.dic_error_api_invalid_email]: 'Invalid Email',
  [TranslationKeys.dic_error_api_user_disabled]: 'User-disabled',
  [TranslationKeys.dic_error_api_user_not_found]: 'User not found',
  [TranslationKeys.dic_error_api_email_already_in_use]: 'Email already in use',
  [TranslationKeys.dic_error_api_operation_not_allowed]:
    'Operation not allowed',
  [TranslationKeys.dic_error_api_weak_password]: 'Weak-password',
  [TranslationKeys.dic_error_api_default]: 'Unknown error',
  [TranslationKeys.auth_email_already_taken]: 'Email is already taken',
  [TranslationKeys.yup_required_field]: 'Required field',
  [TranslationKeys.yup_capitalize_first_letter]:
    'The first letter must be capitalized',
  [TranslationKeys.yup_only_letters]: 'The field should contain only letters',
  [TranslationKeys.yup_valid_email]:
    'Enter a valid email (e.g., user@example.com)',
  [TranslationKeys.yup_domain_name]: 'Email address must contain a domain name',
  [TranslationKeys.yup_length_password]:
    'Password should be of minimum 8 characters length',
  [TranslationKeys.yup_uppercase_letter]:
    'Password must include at least one uppercase letter (A-Z)',
  [TranslationKeys.yup_lowercase_letter]:
    'Password must include at least one lowercase letter (a-z)',
  [TranslationKeys.yup_digit]: 'Password must include at least one digit (0-9)',
  [TranslationKeys.yup_special_character]:
    'Password must include at least one special character (!@#$%^&*)',
  [TranslationKeys.yup_password_mismatch]: 'Passwords must match',
};
