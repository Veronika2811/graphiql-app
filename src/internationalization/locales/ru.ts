import { TranslationKeys } from 'internationalization/TranslationKeys';

export const russianDictionary = {
  [TranslationKeys.signIn]: 'Войти',
  [TranslationKeys.signUp]: 'Зарегистрироваться',
  [TranslationKeys.logOut]: 'Выйти',
  [TranslationKeys.not_found_page]: 'Страница не найдена',
  [TranslationKeys.welcome_page]: 'Приветственная страница',
  [TranslationKeys.main_title]: 'Лучшая IDE для запросов GraphQL',
  [TranslationKeys.main_subtitle]:
    'Приложение является результатом выполнения финального задания на курсе RS School React.',
  [TranslationKeys.welcome_link_editor_heading]:
    'Запрашивайте и изучайте GraphQL API с помощью',
  [TranslationKeys.welcome_link_authorization_heading]: 'для начала',
  [TranslationKeys.about_project_title_1]: 'Язык запросов для вашего API',
  [TranslationKeys.about_project_subtitle_1]:
    'GraphQL — это язык запросов для API и среда выполнения для выполнения этих запросов с использованием существующих данных. GraphQL предоставляет полное и понятное описание данных в вашем API, дает клиентам возможность запрашивать именно то, что им нужно, и не более того, упрощает постепенное развитие API и предоставляет мощные инструменты разработчика.',
  [TranslationKeys.about_project_title_2]:
    'Просите то, что вам нужно, и получите именно это',
  [TranslationKeys.about_project_subtitle_2]:
    'Отправьте запрос GraphQL к своему API и получите именно то, что вам нужно, ни больше, ни меньше. Запросы GraphQL всегда возвращают предсказуемые результаты. Приложения, использующие GraphQL, работают быстро и стабильно, поскольку они контролируют получаемые данные, а не сервер.',
  [TranslationKeys.about_project_title_3]: 'Об RS School',
  [TranslationKeys.about_project_subtitle_3]:
    'RS School — бесплатная образовательная программа, проводимая сообществом разработчиков The Rolling Scopes с 2013 года. Учиться в RS School может каждый, независимо от возраста, профессиональной занятости и места жительства. Наставниками и тренерами школы являются front-end и javascript-разработчики из разных компаний и стран.',
  [TranslationKeys.about_project_alt]: 'Изображение раздела',
  [TranslationKeys.team]: 'Команда',
  [TranslationKeys.team_alt]: 'Изображение профиля',
  [TranslationKeys.editor_search_bar_label]: 'Адрес API',
  [TranslationKeys.editor_button_change_endpoint]: 'Изменить конечную точку',
  [TranslationKeys.konstantin]: 'Константин',
  [TranslationKeys.veranika]: 'Вероника',
  [TranslationKeys.artem]: 'Артём',
  [TranslationKeys.auth_title]: 'Добро пожаловать',
  [TranslationKeys.auth_email]: 'Электронная почта',
  [TranslationKeys.auth_password]: 'Пароль',
  [TranslationKeys.auth_confirm_password]: 'Подтвердите пароль',
  [TranslationKeys.auth_name]: 'Имя',
  [TranslationKeys.variables]: 'Переменные',
  [TranslationKeys.headers]: 'Заголовки',
  [TranslationKeys.editor_message_missing_URL]: 'Введите конечную точку',
  [TranslationKeys.editor_message_incorrect_URL]:
    'Введите правильную конечную точку',
  [TranslationKeys.editor_message_missing_request]: 'Введите запрос',
  [TranslationKeys.editor_message_check_request]: 'Проверьте запрос',
  [TranslationKeys.editor_message_successful_request]: 'Успешный запрос',
  [TranslationKeys.loading]: 'Загрузка',
  [TranslationKeys.error_boundary_message]: 'Упс! Что-то пошло не так!',
  [TranslationKeys.button_execute_query]: 'Выполнить запрос',
  [TranslationKeys.button_make_beautiful]: 'Сделать красиво',
  [TranslationKeys.auth_success_sign_in]: 'Вы успешно вошли в систему',
  [TranslationKeys.auth_success_sign_up]: 'Регистрация прошла успешно',
  [TranslationKeys.dic_error_api_wrong_password]: 'Неправильный пароль',
  [TranslationKeys.dic_error_api_invalid_email]:
    'Неверный адрес электронной почты',
  [TranslationKeys.dic_error_api_user_disabled]: 'Отключен пользователь',
  [TranslationKeys.dic_error_api_user_not_found]: 'Пользователь не найден',
  [TranslationKeys.dic_error_api_email_already_in_use]:
    'Этот электронный адрес уже занят',
  [TranslationKeys.dic_error_api_operation_not_allowed]:
    'Операция не разрешена',
  [TranslationKeys.dic_error_api_weak_password]: 'Слабый пароль',
  [TranslationKeys.dic_error_api_default]: 'Неизвестная ошибка',
  [TranslationKeys.auth_email_already_taken]:
    'Электронная почта уже существует',
  [TranslationKeys.yup_required_field]: 'Обязательное поле',
  [TranslationKeys.yup_capitalize_first_letter]:
    'Первая буква должна быть заглавной',
  [TranslationKeys.yup_only_letters]: 'Поле должно содержать только буквы',
  [TranslationKeys.yup_valid_email]:
    'Введите действительный адрес электронной почты (например, user@example.com)',
  [TranslationKeys.yup_domain_name]:
    'Адрес электронной почты должен содержать доменное имя',
  [TranslationKeys.yup_length_password]:
    'Длина пароля должна быть минимум 8 символов',
  [TranslationKeys.yup_uppercase_letter]:
    'Пароль должен содержать хотя бы одну заглавную букву (A-Z)',
  [TranslationKeys.yup_lowercase_letter]:
    'Пароль должен содержать хотя бы одну строчную букву (a-z)',
  [TranslationKeys.yup_digit]:
    'Пароль должен содержать хотя бы одну цифру (0–9)',
  [TranslationKeys.yup_special_character]:
    'Пароль должен содержать хотя бы один специальный символ (!@#$%^&*)',
  [TranslationKeys.yup_password_mismatch]: 'Пароли должны совпадать',
};
