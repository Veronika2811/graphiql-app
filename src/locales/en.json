{
  "main_heading": "The Best IDE for GraphQL Requests",
  "main_subtitle": "created within RSSchool",
  "heading": "to get started",
  "heading_link": "Sign In",
  "heading_1": "Query and Explore GraphQL API with",
  "heading_link_1": "GraphiQL IDE",
  "about_title": "A query language for your API",
  "about_subtitle": "GraphQL is a query language for APIs and a runtime for fulfilling those queries with your existing data. GraphQL provides a complete and understandable description of the data in your API, gives clients the power to ask for exactly what they need and nothing more, makes it easier to evolve APIs over time, and enables powerful developer tools.",
  "about_title_1": "Ask for what you need,get exactly that",
  "about_subtitle_1": "Send a GraphQL query to your API and get exactly what you need, nothing more and nothing less. GraphQL queries always return predictable results. Apps using GraphQL are fast and stable because they control the data they get, not the server.",
  "team": "Team",
  "Konstantin": "Konstantin",
  "Veranika": "Veranika",
  "Artem": "Artem",
  "signIn": "Sign In",
  "signUp": "Sign Up",
  "signOut": "Sign Out",
  "welcome": "Welcome",
  "placeholder_email": "Email",
  "placeholder_password": "Password",
  "placeholder_editor": "Please enter  URL API",
  "tooltip_documentation": "Show Documentation Explorer",
  "tooltip_execute_query": "Execute query",
  "button_change_endpoint": "Change Endpoint",
  "tab_variables": "Variables",
  "tab_headers": "Headers"
}
