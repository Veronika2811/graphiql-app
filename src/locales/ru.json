{
  "signIn": "Войти",
  "signUp": "Зарегистрироваться",
  "not_found_page": "Страница не найдена",
  "welcome_page": "Приветственная страница",
  "main_title": "Лучшая IDE для запросов GraphQL",
  "main_subtitle": "Приложение является результатом выполнения финального задания на курсе RS School React.",
  "welcome_link_editor_heading": "Запрашивайте и изучайте GraphQL API с помощью",
  "welcome_link_authorization_heading": "для начала",
  "about_project_title_1": "Язык запросов для вашего API",
  "about_project_subtitle_1": "GraphQL — это язык запросов для API и среда выполнения для выполнения этих запросов с использованием существующих данных. GraphQL предоставляет полное и понятное описание данных в вашем API, дает клиентам возможность запрашивать именно то, что им нужно, и не более того, упрощает постепенное развитие API и предоставляет мощные инструменты разработчика.",
  "about_project_title_2": "Просите то, что вам нужно, и получите именно это",
  "about_project_subtitle_2": "Отправьте запрос GraphQL к своему API и получите именно то, что вам нужно, ни больше, ни меньше. Запросы GraphQL всегда возвращают предсказуемые результаты. Приложения, использующие GraphQL, работают быстро и стабильно, поскольку они контролируют получаемые данные, а не сервер.",
  "about_project_title_3": "Об RS School",
  "about_project_subtitle_3": "RS School — бесплатная образовательная программа, проводимая сообществом разработчиков The Rolling Scopes с 2013 года. Учиться в RS School может каждый, независимо от возраста, профессиональной занятости и места жительства. Наставниками и тренерами школы являются front-end и javascript-разработчики из разных компаний и стран.",
  "about_project_alt": "Изображение раздела",
  "team": "Команда",
  "team_alt": "Изображение профиля",
  "editor_search_bar_label": "Адрес API",
  "editor_button_change_endpoint": "Изменить конечную точку",
  "konstantin": "Константин",
  "veranika": "Вероника",
  "artem": "Артём"
}
