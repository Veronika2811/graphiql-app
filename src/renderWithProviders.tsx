import { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import { render, type RenderOptions } from '@testing-library/react';
import { SnackbarProvider } from 'context/snackbar-provider';
import { LocaleProvider } from 'internationalization/LocaleProvider';
import { MuiProvider } from 'layouts/mui';
import { AppStore, setupStore } from 'store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  store?: AppStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  { store = setupStore(), ...renderOptions }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren): JSX.Element => (
    <MuiProvider>
      <SnackbarProvider>
        <HashRouter>
          <Provider store={store}>
            <LocaleProvider>{children}</LocaleProvider>
          </Provider>
        </HashRouter>
      </SnackbarProvider>
    </MuiProvider>
  );
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export default renderWithProviders;
