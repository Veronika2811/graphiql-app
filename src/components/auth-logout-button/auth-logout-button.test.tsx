import { fireEvent, waitFor } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import { englishDictionary } from 'internationalization/locales/en';
import { vi } from 'vitest';

import { renderWithProviders } from '../../renderWithProviders';

import { AuthLogoutButton } from '.';

function createMatchMedia(width: number): (query: string) => MediaQueryList {
  return (query: string) =>
    ({
      matches: mediaQuery.match(query, { width }),
      addListener: () => {},
      removeListener: () => {},
    }) as unknown as MediaQueryList;
}

const mocks = vi.hoisted(() => ({
  authService: {
    logout: vi.fn(),
  },
}));

vi.mock('api/apiAuthFirebase', () => ({
  __esModule: true,
  default: mocks.authService,
}));

describe('AuthLogoutButton component', () => {
  window.matchMedia = createMatchMedia(1440);

  it('should renders correctly AuthLogoutButton component', () => {
    const container = renderWithProviders(<AuthLogoutButton />);
    expect(container).toMatchSnapshot();
  });

  it('should logOut', async () => {
    const { getByRole } = renderWithProviders(<AuthLogoutButton />);

    const logOutButton = getByRole('button', {
      name: englishDictionary.logOut,
    });

    await waitFor(() => {
      expect(logOutButton).toBeEnabled();
    });

    fireEvent.click(logOutButton);

    await waitFor(() => {
      expect(mocks.authService.logout).toHaveBeenCalled();
    });
  });
});
