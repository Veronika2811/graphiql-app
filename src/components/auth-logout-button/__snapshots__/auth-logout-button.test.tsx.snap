// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthLogoutButton component > should renders correctly AuthLogoutButton component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-1u9pwca-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Log Out
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-1u9pwca-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Log Out
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
